crates/ruff_benchmark/benches/red_knot.rs

```diff
@@ -4,73 +4,50 @@ use codspeed_criterion_compat::{criterion_group, criterion_main, BatchSize, Crit
 
 use red_knot_workspace::db::RootDatabase;
 use red_knot_workspace::workspace::WorkspaceMetadata;
+use ruff_benchmark::TestFile;
 use ruff_db::files::{system_path_to_file, vendored_path_to_file, File};
 use ruff_db::parsed::parsed_module;
 use ruff_db::program::{ProgramSettings, SearchPathSettings, TargetVersion};
+use ruff_db::source::source_text;
 use ruff_db::system::{MemoryFileSystem, SystemPath, TestSystem};
 use ruff_db::vendored::VendoredPath;
 use ruff_db::Upcast;
 
-static FOO_CODE: &str = r#"
-import typing
-
-from bar import Bar
-
-class Foo(Bar):
-    def foo() -> object:
-        return "foo"
-
-    @typing.override
-    def bar() -> object:
-        return "foo_bar"
-"#;
-
-static BAR_CODE: &str = r#"
-class Bar:
-    def bar() -> object:
-        return "bar"
-
-    def random(arg: int) -> int:
-        if arg == 1:
-            return 48472783
-        if arg < 10:
-            return 20
-        while arg < 50:
-            arg += 1
-        return 36673
-"#;
-
-static TYPING_CODE: &str = r#"
-def override(): ...
-"#;
-
 struct Case {
     db: RootDatabase,
     fs: MemoryFileSystem,
-    foo: File,
-    bar: File,
-    typing: File,
+    parser: File,
+    re: File,
+    types: File,
     builtins: File,
 }
 
+const TOMLLIB_312_URL: &str = "https://raw.githubusercontent.com/python/cpython/8e8a4baf652f6e1cee7acde9d78c4b6154539748/Lib/tomllib";
+
+fn get_test_file(name: &str) -> TestFile {
+    let path = format!("tomllib/{name}");
+    let url = format!("{TOMLLIB_312_URL}/{name}");
+    TestFile::try_download(&path, &url).unwrap()
+}
+
 fn setup_case() -> Case {
     let system = TestSystem::default();
     let fs = system.memory_file_system().clone();
-    let foo_path = SystemPath::new("/src/foo.py");
-    let bar_path = SystemPath::new("/src/bar.py");
-    let typing_path = SystemPath::new("/src/typing.pyi");
+    let parser_path = SystemPath::new("/src/parser.py");
+    let re_path = SystemPath::new("/src/_re.py");
+    let types_path = SystemPath::new("/src/_types.py");
```

Comment:
Is there a reason why we're excluding `__init__.py`? It might be nice to directly mimic the normal structure of a Python workspace here, where you'd have the package name nested inside the `src` directory:

```suggestion
    let init_path = SystemPath::new("/src/tomllib/__init__.py");
    let parser_path = SystemPath::new("/src/tomllib/parser.py");
    let re_path = SystemPath::new("/src/tomllib/_re.py");
    let types_path = SystemPath::new("/src/tomllib/_types.py");
```